From 23e9d18fe81552cf88652bc91886cf9cbb27c7f1 Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Tue, 23 Jan 2018 09:55:46 +0000
Subject: [PATCH] Drop Printers panel

The printers panel has a hard requirement on Samba, which is not
available on the Yocto base we use for Continuous.
---
 meson.build             | 20 --------------------
 panels/meson.build      |  1 -
 shell/cc-panel-loader.c |  2 --
 3 files changed, 23 deletions(-)

diff --git a/meson.build b/meson.build
index cd3e9c0db..d271de1ef 100644
--- a/meson.build
+++ b/meson.build
@@ -125,26 +125,6 @@ common_deps = [
   dependency('gtk+-3.0', version: '>= 3.22.20')
 ]
 
-# Check for CUPS 1.4 or newer
-cups_dep = dependency('cups', version : '>= 1.4', required: false)
-assert(cups_dep.found(), 'CUPS 1.4 or newer not found')
-
-# https://bugzilla.gnome.org/show_bug.cgi?id=696766
-cups_cflags = []
-if cups_dep.version().version_compare('>= 1.6')
-  cups_cflags += '-D_PPD_DEPRECATED=""'
-endif
-
-# cups headers
-check_headers = [
-  ['HAVE_CUPS_CUPS_H', 'cups/cups.h'],
-  ['HAVE_CUPS_PPD_H', 'cups/ppd.h']
-]
-
-foreach header: check_headers
-  assert(cc.has_header(header[1], args: cups_cflags), 'CUPS headers not found: ' + header[1])
-endforeach
-
 # Optional dependency for the user accounts panel
 enable_cheese = get_option('cheese')
 if enable_cheese
diff --git a/panels/meson.build b/panels/meson.build
index 048980927..3c56f4de2 100644
--- a/panels/meson.build
+++ b/panels/meson.build
@@ -11,7 +11,6 @@ panels = [
   'notifications',
   'online-accounts',
   'power',
-  'printers',
   'privacy',
   'region',
   'search',
diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index 675833c12..b83ee59e1 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -48,7 +48,6 @@ extern GType cc_wifi_panel_get_type (void);
 extern GType cc_notifications_panel_get_type (void);
 extern GType cc_goa_panel_get_type (void);
 extern GType cc_power_panel_get_type (void);
-extern GType cc_printers_panel_get_type (void);
 extern GType cc_privacy_panel_get_type (void);
 extern GType cc_region_panel_get_type (void);
 extern GType cc_search_panel_get_type (void);
@@ -93,7 +92,6 @@ static struct {
   PANEL_TYPE("notifications",    cc_notifications_panel_get_type),
   PANEL_TYPE("online-accounts",  cc_goa_panel_get_type          ),
   PANEL_TYPE("power",            cc_power_panel_get_type        ),
-  PANEL_TYPE("printers",         cc_printers_panel_get_type     ),
   PANEL_TYPE("privacy",          cc_privacy_panel_get_type      ),
   PANEL_TYPE("region",           cc_region_panel_get_type       ),
   PANEL_TYPE("search",           cc_search_panel_get_type       ),
-- 
2.14.3

